apps/backend/src/libs/plugins/authorization.plugin.ts

```diff
@@ -0,0 +1,70 @@
+import { type FastifyRequest } from "fastify";
+import fp from "fastify-plugin";
+
+import { ExceptionMessage, FastifyHook } from "~/libs/enums/enums.js";
+import { HTTPCode, HTTPHeader } from "~/libs/modules/http/http.js";
+import { AuthError } from "~/modules/auth/libs/exceptions/exceptions.js";
+import { type UserService } from "~/modules/users/users.js";
+
+import { token as jwtToken } from "../modules/token/token.js";
+import { checkIfWhiteRoute } from "./libs/helpers/helpers.js";
+
+type Options = {
+	services: {
+		userService: UserService;
+	};
```

Comment:
```suggestion
	services: {
		userService: UserService;
	}
	whiteRoutes
```
pass whiteRoutes via options to plugin